### ISSUE TITLE
Fix Piwik segments (Use Custom Variables)

### ISSUE THREAD

**Issue by javedyousafzai** [humanitarianresponse/site#138]
Need to fix the current segments in Piwik;

Take out the ones maked with '+'. 

@pcambra, please add more comments regarding the 'Custom variables'.

**Comment by pcambra** [comment]
We need to tweak all the country based segments as they are now looking for a URL (operations/operation, i.e. operations/afghanistan) and they should use our custom variables to track which space is the visitor using, that would be more accurate and correspond with what we see on the Analytics link in the main website.

**Comment by pcambra** [comment]
[TL;DR] (https://en.wikipedia.org/wiki/Wikipedia:Too_long;_didn't_read) We've been calculating stats wrongly for operations from the beginning.

@MartinMaritim this transformation hasn't been done correctly. If you check the widgets, and the old operations, we're using the contains operator (=@) whereas now is a exact match (==).

You can check the widgets we're using in the analytics page: https://www.humanitarianresponse.info/en/group/node/85/admin/analytics/visitors

However, I think we're calculating wrongly the stats for clusters in general, as if we do contains (=@) for Senegal (space id 41), anything that contains 41 will be a match, so for example the Humanitarian Programme Cycle has id 1141 (https://www.humanitarianresponse.info/node/1141) and it will be counted as visits for Senegal.

See more details on the Piwik admin page: 
https://stats.humanitarianresponse.info/index.php?module=CoreHome&action=index&date=yesterday&period=day&idSite=14#/module=CustomVariables&action=menuGetCustomVariables&date=yesterday&period=day&idSite=14

The only fix for this is to add 2 conditions per space, both in the Piwik UI and the widgets, take Nepal for example, id 85.

Space id (cluster or operation) is exactly "85" 
OR
Space id contains "85;"

But we would face the same issue again with the contains "85;", there's a new feature to add filters that "start with" but that's targeted to 2.15.1 so until then we cannot really fix this, see https://github.com/piwik/piwik/issues/8076

Additionally, I think these ones are either pending to move to operation id or to delete: 
pageUrl=@operations%2Fmyanmar  
pageUrl=@operations%2Fbrazil
pageUrl=@operations%2Fsouth-africa
pageUrl=@operations%2Fsouth-sudan

Which throws the question: What Operations/Clusters should we showing the stats for? Now we're reaching everything in Piwik but I guess it is a given state in HRInfo instead what should be checked.

On top of all that, the downloads haven't been calculated right either, the method used only looked into the top downloads and discarded the long tail. Also, at some point we changed the urls or the scheme of the files (public/private) so the assumptions made on where a file might be aren't accurate.

*In summary*, this has both break our current stats calculation but also revealed many issues and wrong assumptions that were done when it was built, for example, passing a numeric value instead of a unique name for calculating the stats.

I've been working to get this stats right for a couple of days, but every time I fix something, a new problem arises. I think we should return to the whiteboard and fix this from there.

Including @andrejverity and @Yaelle01 for input.

**Comment by Yaelle01** [comment]
@pcambra , let's leave it for the time being (unless there is a minor fix that would improve).  We'd like Emma to flush out the actual requirements for the stats, so that you can look at how to best implement.

FYI @andrejverity

**Comment by pcambra** [comment]
@Yaelle01 bear in mind that this means we're not generating the monthly stats report for November until much later.

**Comment by Yaelle01** [comment]
@Pedro, if the stats were misleading already, I don't see a point in continuing reporting on these.

**Comment by MartinMaritim** [comment]
@pcambra concerning the question "Which throws the question: What Operations/Clusters should we showing the stats for?" - I believe we should show stats for all active operations and clusters.

**Comment by emmajane** [comment]
To follow up on @Yaelle01's comment: I've started a discussion with Patrick regarding who uses the stats, and what we need to be tracking. I will follow up with him next week when I'm in Geneva. 

In the mean time, #211 can be started so that we are better able to evaluate the best system for our needs.
